lib/web_ui/lib/src/engine/util.dart

```diff
@@ -885,3 +885,35 @@ String tileModeString(ui.TileMode tileMode) {
       return 'decal';
   }
 }
+
+/// A size where both the width and height are integers.
+class BitmapSize {
+  const BitmapSize(this.width, this.height);
+
+  /// Returns a [BitmapSize] by rounding the width and height of a [ui.Size] to
+  /// the nearest integer.
+  BitmapSize.fromSize(ui.Size size)
+      : width = size.width.round(),
+        height = size.height.round();
```

Comment:
I'm still not 100% that `round()` is the right choice. If it rounds down, we may see a blank line of pixels in some cases. However, it's worth a shot as an initial attempt (and I'm not clear on the exact situations that could trigger the bad case). The good news is that all the rounding logic is in one place now, so we can adjust it later easily, if need be.

Reply:
ack